inc/Location.hpp

```diff
@@ -0,0 +1,24 @@
+#ifndef LOCATION_HPP
+#define LOCATION_HPP
+
+#include <iostream>
+#include <map>
+#include <vector>
+
+class Location {
+public:
+    Location();
+    ~Location();
+
+private:
+    std::string m_path;
+    std::string m_root;
```

Comment:
I would suggest to init with a default value, where there is one in nginx (or even always?)

For example: `std::string m_root = "html";`
I think this can prevent some headaches while parsing, because you don't need to explicitly set them to default, since they are set anyway